lib/migrate.js

```diff
@@ -26,9 +26,14 @@ function migrate (set, direction, migrationName, fn) {
     }
 
     set.emit('migration', migration, direction)
-    migration[direction](function (err) {
-      if (err) return fn(err)
+    let migrationCompleted = false
 
+    let migrationPromise = migration[direction](function (err) {
+      if (migrationCompleted) {
```

Comment:
Even though I prefer multi-line if statements with curlys, the majority of the code base is done on a single line.  Please follow the common formatting.